homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -196,11 +199,14 @@ def fan_speed_list(self):
     @property
     def device_state_attributes(self):
         """Return the specific state attributes of this vacuum cleaner."""
+
         attrs = {}
         if self.vacuum_state is not None:
             attrs.update({
                 ATTR_DO_NOT_DISTURB:
-                    STATE_ON if self.vacuum_state.dnd else STATE_OFF,
+                    STATE_ON if self.dnd_state.enabled else STATE_OFF,
+                ATTR_DO_NOT_DISTURB_START: str(self.dnd_state.start),
+                ATTR_DO_NOT_DISTURB_END: str(self.dnd_state.end),
```

Comment:
As in `python-miio` v0.2.0, `self.dnd_state` is a list with one dict, I suppose with the new version that would be an object. If not, this won't work.